daffodil-lib/src/main/scala/org/apache/daffodil/lib/exceptions/Assert.scala

```diff
@@ -91,6 +91,12 @@ object Assert extends Assert {
   def usage(testAbortsIfFalse: Boolean, message: String): Unit = macro AssertMacros.usageMacro2
   def usage(testAbortsIfFalse: Boolean): Unit = macro AssertMacros.usageMacro1
 
+  /**
+   * Test for the validity of the state of the API functions
+   */
+  def illegalState(testThrowsIfFalse: Boolean, message: String, cause: Throwable): Unit =
```

Comment:
Do we have other illegal state checks than just the isError issue? 
I suggest we replace this with checkNotError().